server/cluster.go

```diff
@@ -705,104 +706,96 @@ func (c *RaftCluster) putConfig(meta *metapb.Cluster) error {
 	return c.cachedCluster.putMeta(meta)
 }
 
-// GetNamespaces returns all the namespace in etc.d
-func (c *RaftCluster) GetNamespaces() []*Namespace {
-	return c.cachedCluster.getNamespaces()
+// GetNamespaceClassifier returns current namespace classifier.
+func (c *RaftCluster) GetNamespaceClassifier() namespace.Classifier {
+	c.RLock()
+	defer c.RUnlock()
+	if c.coordinator != nil {
+		return c.coordinator.classifier
+	}
+	return nil
 }
 
-// CreateNamespace creates a new Namespace
-func (c *RaftCluster) CreateNamespace(name string) error {
-	c.Lock()
-	defer c.Unlock()
-
-	r := regexp.MustCompile(`^\w+$`)
-	matched := r.MatchString(name)
-	if !matched {
-		return errors.New("Name should be 0-9, a-z or A-Z")
+// GetNamespaces returns all the namespace in etcd.
+// FIXME: remove it after refactor namespace API.
+func (c *RaftCluster) GetNamespaces() []*Namespace {
+	classifier := c.GetNamespaceClassifier()
+	if classifier == nil {
+		return nil
 	}
-
-	cachedCluster := c.cachedCluster
-	if n := cachedCluster.getNamespace(name); n != nil {
-		return errors.New("Duplicate namespace Name")
+	tableClassifier, ok := classifier.(*tableNamespaceClassifier)
```

Comment:
why not using `classifier.GetNamespaces()` directly?

Reply:
`classifier.GetNamespaces()` returns `[]string`, while we need `[]*Namespace` here.